History/HistoryController.js

```diff
@@ -0,0 +1,37 @@
+const historyService = require('./HistoryService.js');
+
+const addAttempt = async (req, res) => {
+  try {
+    const { userId, questionId, code } = req.body;
+    await historyService.addAttempt(userId, questionId, code);
+    res.json({ message: 'SUCCESS: History attempt added' });
+  } catch (err) {
+    res.status(err?.status || 400).json({ error: err?.message || err });
+  }
+};
+
+const deleteUserAttempts = async (req, res) => {
+  try {
+    const { userId } = req.body;
+    await historyService.deleteUserAttempts(userId);
+    res.json({ message: 'SUCCESS: History attempts deleted' });
+  } catch (err) {
+    res.status(err?.status || 400).json({ error: err?.message || err });
+  }
+};
```

Comment:
Should user be allowed to delete their attempts?

Reply:
This is to prepare when a user deregisters their account. Then all records belonging to the user should also be wiped off.

User service will send a notification for History service to call this function.